src/MyConnection.h

```diff
@@ -18,7 +18,8 @@ class MyConnection : boost::noncopyable {
 
 public:
 
-  MyConnection(std::string host, std::string user, std::string password,
+  MyConnection(std::string host, std::string user,
+               Rcpp::Nullable < std::string > password,
```

Comment:
Except for `unix_socket`...

I'd rather have the `dbConnect()` have NULL for all of its arguments and then supply the defaults here.